src/sagemaker_containers/transformers.py

```diff
@@ -0,0 +1,160 @@
+
+            When an InvokeEndpoint operation is made against an Endpoint running SageMaker model server,
+            the model server receives two pieces of information:
+
+                - The request Content-Type, for example "application/json"
+                - The request data content, which is at most 5 MB (5 * 1024 * 1024 bytes) in size.
+
+            The input_fn is responsible to take the request data and pre-process it before prediction.
+
+        Args:
+            input_data (obj): the request data content.
+            content_type (str): the request Content-Type.
+
+        Returns:
+            (obj): data ready for prediction.
+        """
+        return serializers.loads(input_data, content_type)
+
+    def predict_fn(self, model, data):
+        """Function responsible for model predictions.
+
+        Args:
+            model (obj): model loaded by model_fn
+            data: de-serializes data returned by input_fn
+
+         Returns:
+             (obj): data ready for prediction.
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def output_fn(prediction, accept):
+        """Function responsible to serialize the prediction for the response.
+
+        Args:
+            prediction (obj): prediction returned by predict_fn .
+            accept (str): accept content-type expected by the client.
+
+        Returns:
+            (TransformSpec): a namedtuple with the following args:
+
+                * Args:
+                    serialized_prediction: the serialized data to return
+                    accept: the content-type that the data was transformed to.
+        """
+        serialized_prediction = serializers.dumps(prediction, accept)
+        return TransformSpec(serialized_prediction, accept)
+
+    def transform_fn(self, model, input_data, content_type, accept):
+        """Function responsible for input processing, prediction, and output processing.
+
+        Args:
+            model (obj): model loaded by model_fn
+            input_data (obj): the request data content.
+            content_type (str): the request Content-Type.
+            accept: the content-type that the data was transformed to.
+
+        Returns:
+
+        """
+        data = self.input_fn(input_data=input_data, content_type=content_type)
+        prediction = self.predict_fn(model=model, data=data)
+        return self.output_fn(prediction=prediction, accept=accept)
+
+    def initialize(self):
+        """function is called when the worker starts the Flask application.
+
+        It does not have return type or arguments.
+        """
+        self._model = self.model_fn(model_dir=env.ServingEnv().model_dir)
+
+    def transform(self):
+        """Responsible to make predictions against the model.
+
+        Returns:
+            (sagemaker_containers.transformers.TransformSpec): named tuple with prediction data.
+        """
+        request_env = env.RequestEnv()
```

Comment:
again, this is really just Request not RequestEnv

Reply:
Agreed.